crates/shared/src/balancer_sor_api.rs

```diff
@@ -109,20 +131,22 @@ pub struct Swap {
     pub user_data: Bytes,
 }
 
-/// Module for deserializing the "wrapped" big integer types in the Balancer SOR
-/// API response (of the form `{ "type": "BigNumber", hex: "0x..." }`).
-mod serde_u256_wrapped {
+/// Module for deserializing the "wrapped" signed big integer types in the
+/// Balancer SOR API response (`{ "type": "BigNumber", hex: "0x..." }`).
+mod wrapped_big_integer {
```

Comment:
Surely a great nice to have! I wonder if something like this could live more prominently in a web3 crate or `primitive_types` at some point.

Reply:
This is very specific to Balancer SOR. They return numbers like:
```json
{
  "foo": {
    "type": "BigNumber",
    "hex": "-0x1337"
  }
}
```

When the more expected way would be:
```json
{
  "foo": "-0x1337"
}
```

I wouldn't move this out for that reason.